libs/providers/ofrep-web/src/lib/ofrep-web-provider.ts

```diff
@@ -0,0 +1,309 @@
+   */
+  async initialize?(context?: EvaluationContext | undefined): Promise<void> {
+    try {
+      this._context = context;
+      await this._evaluateFlags(context);
+
+      if (this._pollingInterval > 0) {
+        this.startPolling();
+      }
+
+      this._logger?.debug(`${this.metadata.name} initialized successfully`);
+    } catch (error) {
+      if (error instanceof OFREPApiUnauthorizedError || error instanceof OFREPForbiddenError) {
+        throw new ProviderFatalError('Initialization failed', error);
+      }
+      throw error;
+    }
+  }
+
+  /**
+   * Evaluate is a function retrieving the value from a flag in the cache.
+   * @param flagKey - name of the flag to retrieve
+   * @param type - type of the flag
+   * @private
+   */
+  private evaluate<T extends FlagValue>(flagKey: string, type: string): ResolutionDetails<T> {
+    const resolved = this._cache[flagKey];
+    if (!resolved) {
+      throw new FlagNotFoundError(`flag key ${flagKey} not found in cache`);
+    }
+
+    if (isResolutionError(resolved)) {
+      switch (resolved.errorCode) {
+        case EvaluationFailureErrorCode.FlagNotFound:
+          throw new FlagNotFoundError(`flag key ${flagKey} not found: ${resolved.errorDetails}`);
+        case EvaluationFailureErrorCode.TargetingKeyMissing:
+          throw new TargetingKeyMissingError(`targeting key missing for flag key ${flagKey}: ${resolved.errorDetails}`);
+        case EvaluationFailureErrorCode.InvalidContext:
+          throw new InvalidContextError(`invalid context for flag key ${flagKey}: ${resolved.errorDetails}`);
+        case EvaluationFailureErrorCode.ParseError:
+          throw new ParseError(`parse error for flag key ${flagKey}: ${resolved.errorDetails}`);
+        case EvaluationFailureErrorCode.General:
+        default:
+          throw new GeneralError(
+            `general error during flag evaluation for flag key ${flagKey}: ${resolved.errorDetails}`,
+          );
+      }
+    }
+
+    if (typeof resolved.value !== type) {
+      throw new TypeMismatchError(`flag key ${flagKey} is not of type ${type}`);
+    }
+
+    return {
+      variant: resolved.variant,
+      value: resolved.value as T,
+      flagMetadata: resolved.flagMetadata,
+      errorCode: resolved.errorCode,
+      errorMessage: resolved.errorMessage,
+      reason: resolved.reason,
+    };
+  }
+
+  /**
+   * Start polling for flag updates, it will call the bulk update function every pollInterval
+   * @private
+   */
+  private startPolling() {
+    this._pollingIntervalId = setInterval(async () => {
+      try {
+        const now = new Date();
+        if (this._retryPollingAfter !== undefined && this._retryPollingAfter > now) {
+          return;
+        }
+        const res = await this._evaluateFlags(this._context);
+        if (res === EvaluationStatus.SUCCESS_WITH_CHANGES) {
+          this.events?.emit(ClientProviderEvents.ConfigurationChanged, { message: 'Flags updated' });
```

Comment:
It would be nice to include the list of flags that were changed. In flagd, the changes were calculated like [this](https://github.com/open-feature/js-sdk-contrib/blob/9d6fe63e8954fa56a5ed3e3fdf3117b835ab1bc1/libs/shared/flagd-core/src/lib/storage.ts).